inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/repositories/ReadyForDisplayMessageRepository.kt

```diff
@@ -30,11 +30,15 @@ internal abstract class ReadyForDisplayMessageRepository : ReadyMessageRepositor
 
         override fun getAllMessagesCopy(): List<Message> = ArrayList(messages)
 
-        override fun removeMessage(campaignId: String) {
+        override fun removeMessage(campaignId: String): Message {
             synchronized(messages) {
+                val message = messages.first { msg ->
```

Comment:
I wonder what's better. 2 loops or an if statement in `messages.removeAll` closure that will assign a value to `message` var if the condition is true

Reply:
removed this loop since no longer necessary